### PR TITLE
INTYGFV-15912: Extracted value converters to separate classes. Refactored conversion logic for readability and modified logic to use id/parent in elements.

### PR DESCRIPTION
**Pull request by mww-sopra-steria**

### CODE REVIEW

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/CertificateElementOrganizer.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    sortQuestionsInIndexOrder(categoryQuestionMap);
+
+    return categoryQuestionMap;
+  }
+
+  private static Map<String, CertificateDataElement> getQuestionMap(
+      List<CertificateDataElement> certificateDataElements) {
+    return certificateDataElements.stream()
+        .filter(Predicate.not(CertificateElementOrganizer::elementIsCategory))
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/CertificateElementOrganizer.java:47]: getQuestionMap and getCategoryMap are almost the same. Could they use common logic, or do you feel like that would make the code harder to read?

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeListValueConverter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                .map(CodeListValueConverter::codeToString)
+                .toList()
+        )
+        .build();
+  }
+
+  private static CertificateQuestionValueList notProvidedValueList() {
+    return CertificateQuestionValueList.builder()
+        .values(
+            List.of(NOT_PROVIDED)
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeListValueConverter.java:59]: Should it be a list value and not a text value if it's "not provided"? In that case we will present it like a ul, li html element

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeValueConverter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        .map(RadioMultipleCodeOptionalDropdown::getLabel)
+        .toList();
+
+    return CertificateQuestionValueText.builder()
+        .value(radioMultipleCodeOptionalDropdownLabel.isEmpty() ? NOT_PROVIDED
+            : radioMultipleCodeOptionalDropdownLabel.get(0))
+        .build();
+  }
+
+  private static CertificateQuestionValueText notProvidedTextValue() {
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeValueConverter.java:37]: The same code is used in several classes. Could we extract it to a factory or do we not want to separate the code from the class?

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/DiagnosisListValueConverter.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        .toList();
+
+    final var values = certificateDataValueDiagnoses.stream()
+        .map(diagnosis -> List.of(diagnosis.getCode(), diagnosis.getDescription()))
+        .toList();
+
+    return buildQuestionValueTable(headings, values);
+  }
+
+  private static CertificateQuestionValueTable buildQuestionValueTable(List<String> headings,
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/DiagnosisListValueConverter.java:37]: This feels like a factory method and not unique for this class

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueConverter.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package se.inera.intyg.minaintyg.integration.webcert.converter.data.value;
+
+import se.inera.intyg.minaintyg.integration.api.certificate.model.value.CertificateQuestionValue;
+import se.inera.intyg.minaintyg.integration.webcert.client.dto.CertificateDataElement;
+import se.inera.intyg.minaintyg.integration.webcert.client.dto.value.CertificateDataValueType;
+
+public interface ValueConverter {
+
+  String NOT_PROVIDED = "Ej angivet";
+  String TECHNICAL_ERROR = "Kan inte visa värdet pga tekniskt fel!";
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueConverter.java:10]: I would remove the !

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueToolkit.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+public final class ValueToolkit {
+
+  private static final String NOT_PROVIDED = "Ej angivet";
+
+  private ValueToolkit() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  public static List<CertificateQuestion> getValueDateListSubQuestions(
```
Comment (mhornfeldt) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueToolkit.java:21]: Will this be extracted into its own converter like for the other value types?

**Review by mhornfeldt — COMMENTED** [review]
Nice! :)

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeListValueConverter.java (hunk 2 above)
Comment (mww-sopra-steria) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeListValueConverter.java]: Fixed!

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/DiagnosisListValueConverter.java (hunk 4 above)
Comment (mww-sopra-steria) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/DiagnosisListValueConverter.java:37]: Agreed! I have consciously left it as is in this PR. When we continue the conversion of value we can improve this.

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueConverter.java (hunk 5 above)
Comment (mww-sopra-steria) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueConverter.java]: Done!

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueToolkit.java (hunk 6 above)
Comment (mww-sopra-steria) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/ValueToolkit.java:21]: Yes, as we discussed we will replace this with a new converter that doesn't create subquestions, but instead a question value type that can cater for the need.

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/CertificateElementOrganizer.java (hunk 1 above)
Comment (mww-sopra-steria) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/CertificateElementOrganizer.java:47]: I decide for now to leave as is... the reason is for simplicity and readability.

**Review by mhornfeldt — APPROVED** [review]

**Pull request merged**